perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java

```diff
@@ -899,4 +897,14 @@ public int getDestinationsCount(PerunSession sess) {
 		}
 	}
 
+	@Override
+	public void deleteDestination(PerunSession sess, Destination destination) throws DestinationAlreadyRemovedException {
+		try {
+			if (0 == jdbc.update("delete from destinations where id = ?", destination.getId())) {
```

Comment:
I wonder, if we could catch `DataIntegrityViolationException` to handle the case of existing relations during the delete operation. Then we could also throw `RelationExistsException` from here. But I'm not sure, its actually thrown by Spring.

Reply:
Yes, `DataIntegrityViolationException` is thrown there by Spring. So I added catch clause for it and `RelationExistsException` is thrown from there.